server/src/main/java/org/elasticsearch/search/fetch/QueryFetchSearchResult.java

```diff
@@ -38,9 +39,10 @@ public QueryFetchSearchResult(StreamInput in) throws IOException {
         fetchResult = new FetchSearchResult(in);
     }
 
-    public QueryFetchSearchResult(QuerySearchResult queryResult, FetchSearchResult fetchResult) {
+    public QueryFetchSearchResult(QuerySearchResult queryResult, FetchSearchResult fetchResult, TaskInfo taskInfo) {
         this.queryResult = queryResult;
         this.fetchResult = fetchResult;
+        setTaskInfo(taskInfo);
```

Comment:
We shouldn't set the task info before hand. As explained above,  we want the task info generated at the end of the task so we have to be lazy here and only set the task info when the it is finished (just before sending back the result).